### PR TITLE
feat: add OCI integration resource

### PR DESCRIPTION
**Pull request by kolbeinn**
***Issue***:  [SPM-765](https://lacework.atlassian.net/browse/SPM-765)

***Description:***
This PR adds a resource for OCI cloud integration. It also adds a TF acceptance test and an integration test and updates the go-sdk dependency to use the latest release.

***Additional Info:***
This was tested by running the two included tests. First, the integration test was run as follows:
```
LW_ACCOUNT=$(lacework configure show account) \
LW_API_KEY=$(lacework configure show api_key) \
LW_API_SECRET=$(lacework configure show api_secret) \
OCI_CREDENTIALS=$(jq -c . oci_credentials.json) \
make integration-test regex=Oci
```
where the `oci_credentials.json` file follows the JSON schema defined in `integration/oci_auth.go`. Secondly, I run the TF_ACC test as follows:
```
LW_ACCOUNT=$(lacework configure show account) \
LW_API_KEY=$(lacework configure show api_key) \
LW_API_SECRET=$(lacework configure show api_secret) \
OCI_TENANT_ID=<OCI tenant OD> \
OCI_TENANT_NAME=<OCI tenant name> \
OCI_HOME_REGION=<OCI tenant region> \
OCI_USER_OCID=<OCI user OCID> \
OCI_FINGERPRINT=<OCI public key fingerprint> \
OCI_PRIVATE_KEY=<OCI private key in PEM format> \
TF_ACC=1 \
go test -v ./lacework/... -run TestAccIntegrationOciCfg
```
Both tests passed and I checked the Lacework tenant audit logs to verify that the cloud integrations had actually been created and destroyed during the tests.

[SPM-765]: https://lacework.atlassian.net/browse/SPM-765?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kolbeinn** [comment]
My CI pipeline is failing because I haven't added the OCI_CREDENTIALS to the Codefresh pipeline. Working on it now.

**Comment by kolbeinn** [comment]
I got the OCI credentials added to the pipeline environment. Now I need to add the customer demo environment (the target Lacework environment in the CI pipeline) added to the feature flag for the OCI API functionality.

File: integration/resource_lacework_integration_oci_cfg_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		actualName := terraform.Output(t, terraformOptions, "name")
+		assert.Equal(
+			t,
+			"OciCfg_created_by_terraform",
+			GetCloudAccountIntegrationName(create),
+		)
+		assert.Equal(t, "OciCfg_created_by_terraform", createData.Data.Name)
+		assert.Equal(t, "OciCfg_created_by_terraform", actualName)
+
+		// Update AwsCfg Integration
```
Comment (dmurray-lacework) [inline integration/resource_lacework_integration_oci_cfg_test.go:46]:
```suggestion
		// Update OciCfg Integration
```

**Review by dmurray-lacework — APPROVED** [review]
Looks good, just need to get CI passing

**Comment by dmurray-lacework** [comment]
The OCI test is returning 403
`[403] This feature will be available later.`

**Comment by iok0** [comment]
> The OCI test is returning 403 `[403] This feature will be available later.`

Think it's behind a launch darkly feature flag.

**Comment by iok0** [comment]
Looks good. We should probably santiy check it also works from the cloud shell (where you provide `null` for some of the auth items).
We checked this for the ad hoc terraform recently and added steps to current private preview docs:[terraform-via-oci-cloud-shell
](https://github.com/lacework-dev/scripts/tree/main/lw-oci-integration/tf#prepare-to-run-terraform-via-oci-cloud-shell-option-2)

**Review by iok0 — APPROVED** [review]

**Comment by rmoles** [comment]
@kolbeinn can you provide the CUST GUID for the Lacework account being used here? I'll add it to the Feature Flag

**Review by rmoles — APPROVED** [review]

**Review by afiune — COMMENTED** [review]
**Not a blocker:** @kolbeinn could you please work with @dmurray-lacework and add the credentials used in this integration tests into our Secrets Management store, then add it to https://lacework.atlassian.net/wiki/spaces/ALLY/pages/2309030376/Team+Shared+Credentials

Great work! 💯

**Comment by kolbeinn** [comment]
> **Not a blocker:** @kolbeinn could you please work with @dmurray-lacework and add the credentials used in this integration tests into our Secrets Management store, then add it to https://lacework.atlassian.net/wiki/spaces/ALLY/pages/2309030376/Team+Shared+Credentials
> 
> Great work! 💯

Thanks! Will do

**Comment by kolbeinn** [comment]
> Looks good. We should probably santiy check it also works from the cloud shell (where you provide `null` for some of the auth items). We checked this for the ad hoc terraform recently and added steps to current private preview docs:[terraform-via-oci-cloud-shell ](https://github.com/lacework-dev/scripts/tree/main/lw-oci-integration/tf#prepare-to-run-terraform-via-oci-cloud-shell-option-2)

@iok0 Tested this from the OCI cloud shell and it works just fine

**Comment by kolbeinn** [comment]
@dmurray-lacework @afiune Credentials added to Secrets Manager and wiki page updated

**Comment by afiune** [comment]
@kolbeinn Awesome!

One more thing folks, this PR is adding a new resource so we essentially missed the documentation part, so nothing will show up at https://registry.terraform.io/providers/lacework/lacework/latest/docs until we add something to the `website/` directory.

Here is an example of a PR that added a new resource with docs. cc @dmurray-lacework

**Comment by kolbeinn** [comment]
Got it, @afiune. Created a separate PR with the docs: #509

**Pull request merged**